nob.h

```diff
@@ -94,7 +94,7 @@ bool nob_read_entire_dir(const char *parent, Nob_File_Paths *children);
 bool nob_write_entire_file(const char *path, void *data, size_t size);
 Nob_File_Type nob_get_file_type(const char *path);
 
-#define nob_return_defer(value) do { result = (value); goto defer; } while(0)
+#define nob_return_defer for(;;({ goto defer; })) result =
```

Comment:
Isn't this a GNU statement expression extension? I want nob to be compilable by as many compilers as possible.

Reply:
Changed it to only include standard C stuff.

Reply:
Yeah, noticed the warning too, which was why I included the for loop version when the compiler supports statement expressions. The warning _can_ be turned off locally though with a pragma in the macro since the dangling else warning is bogus in this case.